lib/puppet/provider/package/windows.rb

```diff
@@ -64,12 +64,21 @@ def install
 
     command = [installer.install_command(resource), install_options].flatten.compact.join(' ')
     working_dir = File.dirname(resource[:source])
-    if !Puppet::FileSystem.exist?(working_dir) && resource[:source] =~ /\.msi"?\Z/i
+    unless Puppet::FileSystem.exist?(working_dir)
       working_dir = nil
     end
     output = execute(command, :failonfail => false, :combine => true, :cwd => working_dir, :suppress_window => true)
 
     check_result(output.exitstatus)
+
+    begin
+      if installer == Puppet::Provider::Package::Windows::ExePackage && resource[:source].start_with?('http://', 'https://')
+        downloaded_file_location = File.join(Dir.tmpdir, File.basename(resource[:source]))
+        File.delete(downloaded_file_location) if File.exist?(downloaded_file_location)
```

Comment:
Since `Dir.tmpdir` is often writable by non-privileged users, it's important that the path be unpredictable and not be writable by `other`. I'd just use Tempfile.new as it will handle both.

Reply:
I switched to `Tempfile.new(['','.exe'])`